renderdoc/api/replay/d3d12_pipestate.h

```diff
@@ -436,43 +431,55 @@ struct ConstantBuffer
   rdcarray<uint32_t> rootValues;
 };
 
-DOCUMENT("Contains all of the registers in a single register space mapped to by a root signature.");
-struct RegisterSpace
+DOCUMENT("Contains information for a single root signature element range");
+struct RootElement
```

Comment:
I can appreciate the reasoning for having each range with its own struct, since that's the granularity where register space and shader visibility can change and we expect there to be comparatively few of them, so it avoids needing to check per-entry what the space or visibility is. That's fine, but I think the name `RootElement` is too misleading in that context, since that immediately makes me think that there's one of these for each root signature element. Especially since there's a member `rootElement` which feels like it should be the index in the parent array and it easily could be if multiple ranges aren't used.

Naming discussions are the bane of computer science, but I think we need to clarify that this isn't quite the same. `RootSignatureRange` maybe? Then anyone familiar with the ugly root signature internals will know roughly the granularity to expect, and anyone not familiar will still understand that it's some subrange of the root signature.